src/components/MerchantList.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import Merchant from './Merchant';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+const initialMerchants = [
+  { id: 0, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 1, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 2, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 3, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+  { id: 4, name: "海洋大學店", distance: 1.6, costDownLimit: 98, costUpLimit: 123, starRate: 4.8, starNumber: 71 },
+  { id: 5, name: "海", distance: 2.5, costDownLimit: 12, costUpLimit: 124, starRate: 3.2, starNumber: 12 },
+];
+
+function MerchantList() {
+  const [merchants, setMerchants] = useState(initialMerchants); 
+  const [hasMore, setHasMore] = useState(true); 
+  const [loading, setLoading] = useState(false); 
+  const [isBottom, setIsBottom] = useState(false);
+  const loadMoreMerchants = () => {
+    if (loading || !hasMore) return;
+    setLoading(true);
+    setTimeout(() => {
+      const newMerchants = merchants.concat(
+        Array.from({ length: 3 }).map((_, index) => ({
+          id: merchants.length + index,
+          name: `新商家 ${merchants.length + index}`,
+          distance: (Math.random() * 10).toFixed(1),
+          costDownLimit: Math.floor(Math.random() * 100),
+          costUpLimit: Math.floor(Math.random() * 200),
+          starRate: (Math.random() * 5).toFixed(1),
+          starNumber: Math.floor(Math.random() * 100),
+        }))
+      );
+
+      setMerchants(newMerchants);
+      
+      if (newMerchants.length >= 50) {
+        setHasMore(false);
+      }
+      setLoading(false); 
+      setIsBottom(false);
+    }, 2000);
+  };
+
+  const handleScroll = () => {
```

Comment:
Don't use this method to detect bottom like this.
Use `IntersectionObserver` API to accomplish.
Read this reference:
https://jim1105.coderbridge.io/2022/07/30/intersection-observer/